src/contract.rs

```diff
@@ -110,6 +110,7 @@ impl<'a> AffiliateSwap<'a> {
         let fee_percentage = std::cmp::min(max_fee_percentage, fee_percentage);
 
         // calculate the fee to deduct
+        // (boss): [question] not using checked_mul here as error is unexpected?
```

Comment:
mostly cause it's easier to read without it. I wouldn't expect overflows to be possible, but maybe I'm wrong?

Reply:
I question this for the same reasoning as sub but feel free to make your final judgement :)

Reply:
realized that this becomes harder to write because checked_mul is not defined for <Uint128, Decimal> while Mul is. So you end up with a bunch of boilerplate conversions.